Automattic/wp-calypso
# Add support for `?coupon=[code]` in Calypso.

_This is a fork of https://github.com/Automattic/wp-calypso/pull/25395_

- Minus the cookie and with a few enhancements based on previous feedback.

## Testing Instructions

- Visit: https://calypso.live/?branch=add/coupon-urls-lite
- Add something to your cart.
- From the checkout page where items are visible in your cart, instead of entering a coupon code, add `?coupon=[code]` onto the end of the current URL and reload the page; e.g., `https://calypso.live/checkout/[mytestsite]?coupon=[code]`  
  (replace `[code]` with a valid coupon).
- Confirm the coupon is applied automatically and that your price is discounted.
  <img width="480" alt="2018-06-08_07-28-39" src="https://user-images.githubusercontent.com/1563559/41167008-34af8bac-6aee-11e8-9da0-04876700f225.png">

## TODO

- ~~[ ] Signup flow considerations.~~ To be considered in the future.

---

_This PR is rebased on https://github.com/Automattic/wp-calypso/pull/25439_
